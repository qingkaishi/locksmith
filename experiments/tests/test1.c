#include <pthread.h>
#include <stdio.h>

int *j;

void * dostuff(void * a) {
  int k;
  pthread_mutex_t l;
  
  pthread_mutex_init(&l, NULL);
  pthread_mutex_lock(&l);
  j = &k;
  pthread_mutex_unlock(&l);
  return NULL;
}

int main() {
  pthread_t t1;
  int i = 1;
  pthread_create(&t1, NULL, dostuff, NULL);
  pthread_create(&t1, NULL, dostuff, NULL);
}


