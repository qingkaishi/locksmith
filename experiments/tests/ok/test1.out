************************* STARTING *************************

shared: &j:test1.c:4

Warning: Possible data race: &j:test1.c:4 is not protected!
 references:
  dereference of &j:test1.c:4 at test1.c:12
    &j:test1.c:4
  locks acquired:
    concrete l::-1 -> dostuff test1.c:21
  in: main at test1.c:17 

  dereference of &j:test1.c:4 at test1.c:12
    &j:test1.c:4
  locks acquired:
    concrete l::-1 -> dostuff test1.c:20
  in: main at test1.c:17 


*************************** DONE ***************************
